geth/common/queued_tx.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// QueuedTx holds enough information to complete the queued transaction.
+type QueuedTx struct {
+	id         QueuedTxID
+	hash       common.Hash
+	context    context.Context
+	args       SendTxArgs
+	inProgress bool // true if transaction is being sent
+	done       chan struct{}
+	discard    chan struct{}
+	err        error
+	sync.RWMutex
+}
+
+// NewQueuedTx QueuedTx constructor.
+func NewQueuedTx(id QueuedTxID, ctx context.Context, args SendTxArgs) *QueuedTx {
```

Comment:
Context normally is first argument.

Reply:
It couldnt be done easely `geth/txqueue/txqueue_manager.go:419`
It seems more related to routing issue.